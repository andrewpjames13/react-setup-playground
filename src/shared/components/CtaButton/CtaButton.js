import React, { Component, } from 'react';

class CtaButton extends Component {
  render() {
    return(
      <button
        className={this.props.color}
        onClick={this.props.clickEvent}
        >
          {this.props.buttonText}
        </button>
    );
  }
};

export default CtaButton;
