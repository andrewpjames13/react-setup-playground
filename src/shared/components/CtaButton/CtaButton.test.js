import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import CtaButton from './CtaButton';

function setup(color, buttonText) {
  const props = {
    clickEvent: jest.fn(),
    buttonText,
    color
  }

  const ctaButton = shallow(<CtaButton {...props} />)

  return {
    props,
    ctaButton
  }
}

describe('<CtaButton />', () => {
  it('calls clickEvent on click of button', () => {
    const { ctaButton, props } = setup('primary');

    ctaButton.simulate('click');
    expect(props.clickEvent.mock.calls.length).toBe(1);
  });

  it('renders the correct class', () => {
    const { ctaButton } = setup('primary');

    expect(ctaButton.hasClass('primary')).toBe(true);
  });

  it('renders the correct buttonText', () => {
    const { ctaButton } = setup('primary', 'This is a button');

    expect(ctaButton.find('.primary').text()).toBe('This is a button');
  });

  test('snapshot', () => {
    const { props } = setup('primary', 'This is a button');

    const tree = renderer.create(
      <CtaButton {...props} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
