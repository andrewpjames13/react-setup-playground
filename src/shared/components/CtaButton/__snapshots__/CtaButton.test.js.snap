exports[`<CtaButton /> snapshot 1`] = `
<button
  className="primary"
  onClick={[Function]}>
  This is a button
</button>
`;
